packages/std/src/coins.rs

```diff
@@ -91,8 +92,8 @@ impl fmt::Display for Coins {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let s = self
             .0
-            .iter()
-            .map(|(denom, amount)| format!("{amount}{denom}"))
+            .values()
+            .map(|coin| format!("{coin}"))
```

Comment:
Not sure if it makes a difference but maybe simpler:

```suggestion
            .map(|coin| coin.to_string())
```